PowerShellTools/ServiceManagement/ConnectionManager.cs

```diff
@@ -7,49 +8,87 @@ namespace PowerShellTools.ServiceManagement
     /// <summary>
     /// Manage the process and channel creation.
     /// </summary>
-    internal sealed class ConnectionManager
+    internal static class ConnectionManager
     {
-        private IPowershellIntelliSenseService _powershellIntelliSenseServiceChannel;
-        private int _hostProcessId;
+        private static IPowershellIntelliSenseService _powershellIntelliSenseService;
+        private static int _hostProcessId;
+        private static object _syncObject = new object();
+        private static ChannelFactory<IPowershellIntelliSenseService> _channelFactory;
 
-        public ConnectionManager()
-        {            
+        static ConnectionManager()
+        {
             OpenClientConnection();
         }
 
-        /// <summary>
-        /// The service channel we need.
-        /// </summary>
-        public IPowershellIntelliSenseService PowershellIntelliSenseServiceChannel
+        public static IPowershellIntelliSenseService PowershellIntelliSenseSerivce
         {
             get
             {
-                return _powershellIntelliSenseServiceChannel;
+                if (_powershellIntelliSenseService == null)
+                {
+                    OpenClientConnection();
+                }
+                return _powershellIntelliSenseService;
             }
         }
 
-        private void OpenClientConnection()
+        public static void OpenClientConnection()

```

Comment:
Will fix this

From: Eric Deng [mailto:notifications@github.com]
Sent: Monday, January 26, 2015 11:53 AM
To: Microsoft/poshtools
Cc: Huanhuan Sun
Subject: Re: [poshtools] Add exception handling mechanism for process and service host. (#60)


In PowerShellTools/ServiceManagement/ConnectionManager.cs<https://github.com/Microsoft/poshtools/pull/60#discussion_r23558906>:

>              }

>          }

>

> -        private void OpenClientConnection()

> +        public static void OpenClientConnection()

Does this need to be public?

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/poshtools/pull/60/files#r23558906>.